src/frequenz/sdk/timeseries/__init__.py

```diff
@@ -6,8 +6,35 @@
 
 A timeseries is a stream (normally an async iterator) of
 [`Sample`][frequenz.sdk.timeseries.Sample]s.
+
+# Periodicity and alignment
+
+All the data produced by this package is always periodic and aligned to the
+`UNIX_EPOCH` (by default).
+
+Classes normally take a (re)sampling period as and argument and, optionally, an
+`align_to` argument.
+
+This means timestamps are always separated exaclty by a period, and that this
+timestamp falls always at multiples of the period, starting at the `align_to`.
+
+This ensures that that the data is predictable and consistent among restarts.
+
+Example:
+    If we have a period of 10 seconds, and are aligning to the UNIX
+    epoch. Assuming the following timeline starts in 1970-01-01 00:00:00
+    UTC and our current `now` is 1970-01-01 00:00:32 UTC, then the next
+    timestamp will be at 1970-01-01 00:00:40 UTC:
+
+    ```
+    align_to = 1970-01-01 00:00:00         next event = 1970-01-01 00:00:40
+    |                                       |
+    |---------|---------|---------|-|-------|---------|---------|---------|
+    0        10        20        30 |      40        50        60        70
+                                   now = 1970-01-01 00:00:32
+    ```
```

Comment:
This is jumping a bit ahead and should be true after #343 is done (I'm already working on it). I couldn't find a better place to put this generic documentation on, and the `UNIX_EPOCH`. An `alignment` module doesn't make a lot of sense, as what the ring buffer needs to do is pretty different from what the resampler needs to do, and at the end the alignment itself is just a simple math operation: `now + period - (now - align_to) % period`, a bit too simple to deserve its own module.